### PR TITLE
feat: respect multiple browser languages, refs #2

### PR DESCRIPTION
**Pull request by yamenhadla**

### CODE REVIEW

File: Classes/Middleware/LanguageRedirectionMiddleware.php [hunk 1]
```diff
@@ -30,22 +30,32 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
-        // Get the iso code of the browser language from HTTP request header
-        $browserLanguageIsoCode = substr($request->getServerParams()['HTTP_ACCEPT_LANGUAGE'], 0, 2);
-
-        foreach ($siteLanguages as $siteLanguage) {
-            // Check if the browser language is supported by comparing two letter iso codes
-            if ($browserLanguageIsoCode === $siteLanguage->getTwoLetterIsoCode()) {
-                // Do nothing, if the site language base URL is the currently requested URL
-                if ((string) $siteLanguage->getBase() === $request->getAttribute('normalizedParams')->getRequestUrl()) {
-                    return $handler->handle($request);
+        // Get the iso code of the browser languages from HTTP request header
```
Comment (malteriechmann) [inline Classes/Middleware/LanguageRedirectionMiddleware.php:33]:
```suggestion
        // Get the iso codes of the browser languages from HTTP request header
```

File: Classes/Middleware/LanguageRedirectionMiddleware.php [hunk 2]
```diff
@@ -30,22 +30,32 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
+        $browserLanguageIsoCodes = $this->getBrowserLangugeIsoCodes($request->getServerParams()['HTTP_ACCEPT_LANGUAGE']);
+
+        // Check which of the browser language is supported by comparing two letter iso codes
```
Comment (malteriechmann) [inline Classes/Middleware/LanguageRedirectionMiddleware.php:36]:
```suggestion
        // Check which of the browser languages is supported by comparing two letter iso codes
```

File: Classes/Middleware/LanguageRedirectionMiddleware.php [hunk 3]
```diff
@@ -30,22 +30,32 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
...
-
-                // Redirect the user to the preferred language base URL
-                return new RedirectResponse($siteLanguage->getBase());
             }
         }
 
         return $handler->handle($request);
     }
+
+    protected function getBrowserLangugeIsoCodes($acceptLanguageHeader)
```
Comment (malteriechmann) [inline Classes/Middleware/LanguageRedirectionMiddleware.php:53]: Please add some type hinting to this method.

Comment (Yanohd) [reply Classes/Middleware/LanguageRedirectionMiddleware.php:54]: Resolved on the following comment:

`// This function takes in an Accept-Language header string and returns an array of unique two-letter ISO codes`

Comment (malteriechmann) [reply Classes/Middleware/LanguageRedirectionMiddleware.php:54]: I think the comment is not necessary, but you can add type hinting in TYPO3 source code like this:

```
protected function getBrowserLangugeIsoCodes(string $acceptLanguageHeader): array
```

**Review by malteriechmann — APPROVED** [review]

**Pull request merged**